zebra/rtadv.c

```diff
@@ -632,45 +633,58 @@ static void rtadv_process_advert(uint8_t *msg, unsigned int len,
 
 	radvert = (struct nd_router_advert *)msg;
 
-	if ((radvert->nd_ra_curhoplimit && zif->rtadv.AdvCurHopLimit)
-	    && (radvert->nd_ra_curhoplimit != zif->rtadv.AdvCurHopLimit)) {
+	monotime(&now);
+	past = now;
+	past.tv_sec -= (6 * 60 * 60);
+
+	if ((radvert->nd_ra_curhoplimit && zif->rtadv.AdvCurHopLimit) &&
+	    (radvert->nd_ra_curhoplimit != zif->rtadv.AdvCurHopLimit) &&
+	    timercmp(&past, &zif->rtadv.lastadvcurhoplimit, >)) {
 		flog_warn(
 			EC_ZEBRA_RA_PARAM_MISMATCH,
 			"%s(%u): Rx RA - our AdvCurHopLimit doesn't agree with %s",
 			ifp->name, ifp->ifindex, addr_str);
+		zif->rtadv.lastadvcurhoplimit = now;
 	}
 
-	if ((radvert->nd_ra_flags_reserved & ND_RA_FLAG_MANAGED)
-	    && !zif->rtadv.AdvManagedFlag) {
+	if ((radvert->nd_ra_flags_reserved & ND_RA_FLAG_MANAGED) &&
+	    !zif->rtadv.AdvManagedFlag &&
+	    timercmp(&past, &zif->rtadv.lastadvmanagedflag, >)) {
 		flog_warn(
 			EC_ZEBRA_RA_PARAM_MISMATCH,
 			"%s(%u): Rx RA - our AdvManagedFlag doesn't agree with %s",
 			ifp->name, ifp->ifindex, addr_str);
+		zif->rtadv.lastadvmanagedflag = now;
 	}
 
-	if ((radvert->nd_ra_flags_reserved & ND_RA_FLAG_OTHER)
-	    && !zif->rtadv.AdvOtherConfigFlag) {
+	if ((radvert->nd_ra_flags_reserved & ND_RA_FLAG_OTHER) &&
+	    !zif->rtadv.AdvOtherConfigFlag &&
+	    timercmp(&now, &zif->rtadv.lastadvotherconfigflag, >)) {
```

Comment:
this should be `past`, right? `now` is always going to be greater than the `now` you stored in the past.

Reply:
problem fixed with change in approach